### PR TITLE
[SDAO-299] Add getSwap function

### PR DESCRIPTION
**Pull request by gpevnev**
Problem: Swap information needed to be accessed by backend

Solution: Add `getSwap` view funnction to expose swap information

### CODE REVIEW

File: contracts/BlendSwap.sol [hunk 1]
```diff
@@ -38,6 +38,10 @@ contract BlendSwap {
         blend = IERC20(blend_);
     }
 
+    function getSwap(bytes32 secretHash) public view returns (Swap memory) {
```
Comment (gpevnev) [inline contracts/BlendSwap.sol:41]: @kkirka do you have experience with returning structs from _view functions_? Will this work properly?

Comment (kkirka) [reply contracts/BlendSwap.sol:41]: Wait, why do you even need this function? `mapping (bytes32 => Swap) public swaps` is `public`, so Solidity automatically generates the view of the form `function swaps(bytes32) public view returns (Swap memory)`.

Regarding "returning structs from view functions" – yes, it works fine, see, e.g., https://github.com/StakerDAO/blend/blob/master/test/BlendSwap.js#L90

**Review by kkirka — CHANGES_REQUESTED** [review]
I don't get why it's necessary but if you like `getSwap(...)` more than `swaps(...)`, please make the mapping private and update the tests accordingly

**Comment by gpevnev** [comment]
@kkirka I see, if I would be able to call to `swaps` successfully function from Haskell I would close this PR

**Comment by gpevnev** [comment]
Looking that there are no problems with using autogenerated `swaps` getter for mapping with same name from Haskell code

**Pull request closed without merging**